docs/usage/types/dicts_mapping.md

```diff
@@ -18,3 +61,67 @@
     Prior to Python 3.8, it requires the [typing-extensions](https://pypi.org/project/typing-extensions/) package.
     But required and optional fields are properly differentiated only since Python 3.9.
     We therefore recommend using [typing-extensions](https://pypi.org/project/typing-extensions/) with Python 3.8 as well.
+
+Same as `dict` but Pydantic will validate the dictionary since keys are annotated.
+
+```py
+from typing import Optional
+
+from typing_extensions import TypedDict
+
+from pydantic import BaseModel, ConfigDict, ValidationError
+
+
+# `total=False` means keys are non-required
+class UserIdentity(TypedDict, total=False):
```

Comment:
You can use the `# (1)!` feature of mkdocs-material 👀

Reply:
Apply if you want.

Reply:
please create an issue about this, we should fix it.